storages/request_dispatcher.go

```diff
@@ -14,60 +21,116 @@ type dispatcher interface {
 type RequestDispatcher struct {
 	Backends                  []*backend.Backend
 	syncLog                   *SyncSender
-	pickClientFactory         func(*http.Request) func([]*backend.Backend) client
-	pickResponsePickerFactory func(*http.Request) func(<-chan BackendResponse) responsePicker
+	pickClientFactory         func(*http.Request) func([]*backend.Backend, watchdog.ConsistencyWatchdog) client
+	pickResponsePickerFactory func(*Request) func(<-chan BackendResponse) responsePicker
+	watchdog                  watchdog.ConsistencyWatchdog
+	watchdogRecordFactory     watchdog.ConsistencyRecordFactory
 }
 
 // NewRequestDispatcher creates RequestDispatcher instance
-func NewRequestDispatcher(backends []*backend.Backend, syncLog *SyncSender) *RequestDispatcher {
-
+func NewRequestDispatcher(backends []*backend.Backend, syncLog *SyncSender,
+	watchdog watchdog.ConsistencyWatchdog, watchdogRecordFactory watchdog.ConsistencyRecordFactory) *RequestDispatcher {
 	return &RequestDispatcher{
 		Backends:                  backends,
 		syncLog:                   syncLog,
 		pickResponsePickerFactory: defaultResponsePickerFactory,
 		pickClientFactory:         defaultReplicationClientFactory,
+		watchdog:                  watchdog,
+		watchdogRecordFactory:     watchdogRecordFactory,
 	}
 }
 
 // Dispatch creates and calls replicators and response pickers
 func (rd *RequestDispatcher) Dispatch(request *http.Request) (*http.Response, error) {
+	storageRequest := &Request{request, nil, nil}
+	if rd.shouldUseConsistencyWatchdog(request) {
+		recordedRequest, err := rd.createAndInsertRecordFor(request)
```

Comment:
I think this would be clearer:
var err error
storageRequest, err := rd.createAndInsertRecordFor(request)
than setting properties one by one

Reply:
I don't understand why would we want  to replace `err := ...` with `var err error`. Can you explain?

Reply:
sry, my fault there is no need to declare err separately

Reply:
goal was to eliminate recordedRequest variable